crates/bevy_render/src/world_sync.rs

```diff
@@ -161,11 +165,24 @@ pub(crate) fn entity_sync_system(main_world: &mut World, render_world: &mut Worl
                         };
                     }
                 }
-                EntityRecord::Removed(e) => {
-                    if let Some(ec) = render_world.get_entity_mut(e) {
+                EntityRecord::Removed(render_entity) => {
+                    if let Some(ec) = render_world.get_entity_mut(render_entity.id()) {
                         ec.despawn();
                     };
                 }
+                EntityRecord::ComponentRemoved(main_entity) => {
+                    // It's difficult to remove only the relevant component because component ids aren't stable across worlds,
+                    // so we just clear the entire render world entity.
```

Comment:
This is fun and has consequences, but we're gonna figure that out.

Reply:
After thinking more I think it's what we need to do in any case. Components like `SpotLight` aren't extracted directly but generate derived values, e.g. `ExtractedPointLight`. The only reason `SyncComponentPlugin` works in this case is because we clear everything.

Reply:
I'm deferring on your judgement on this as I don't have much insight in rendering internals. As long as it's correctly documented (and maybe also include I reason why we have to do this, besides it being difficult, because it's strictly not impossible), this should be good.

Reply:
@IceSentry or another SME should perhaps sanity check this logic, I'm not an expert in the internals either but it makes sense to me :D